frontend/common/src/components/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,249 @@
+    Scalars["ID"] | null
+  >(null);
+  const [query, setQuery] = React.useState<string>("");
+  const {
+    setValue,
+    resetField,
+    register,
+    formState: { errors },
+  } = useFormContext();
+  const inputProps = register(name, rules);
+  const stateStyles = useFieldStateStyles(name, !trackUnsaved);
+  const fieldState = useFieldState(name || "", !trackUnsaved);
+  const isUnsaved = fieldState === "dirty" && trackUnsaved;
+  const error = errors[name]?.message as FieldError;
+  const isRequired = !!rules?.required;
+
+  // TODO: Make this filter much smarter, possibly fuse.js
+  const filteredOptions = React.useMemo(() => {
+    if (query === "" || isExternalSearch) {
+      return options;
+    }
+
+    return options.filter((option) =>
+      option.label
+        ?.toLocaleString()
+        .toLowerCase()
+        .includes(query.toLowerCase()),
+    );
+  }, [query, isExternalSearch, options]);
+
+  const noOptions = fetching || filteredOptions.length === 0;
+
+  const helpId = `${id}-error`;
+
+  const handleChange = React.useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const {
+        target: { value },
+      } = e;
+      setQuery(value);
+      if (onSearch) {
+        onSearch(value);
+      }
+    },
+    [onSearch],
+  );
+
+  const debouncedChangeHandler = React.useMemo(
+    () => debounce(handleChange, 300),
+    [handleChange],
+  );
+
+  React.useEffect(() => {
+    return () => {
+      debouncedChangeHandler.cancel();
+    };
+  }, [debouncedChangeHandler]);
+
+  const handleClear = () => {
+    setSelectedOption(null);
+    setQuery("");
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  };
+
+  // This does not play well with react-hook-form so
+  // we need to manually set the value here
+  React.useEffect(() => {
+    if (selectedOption !== "") {
+      setValue(name, selectedOption);
+    } else {
+      resetField(name);
+    }
+  }, [selectedOption, setValue, resetField, name]);
+
+  const getDisplayValue = (value: Scalars["ID"]): string => {
+    if (value) {
+      const selected = options.find((option) => option.value === value);
+      if (selected) {
+        return `${selected.label}`;
+      }
+    }
+
+    return ``;
+  };
+
+  return (
+    <ComboboxPrimitive
+      as="div"
+      value={selectedOption}
+      onChange={setSelectedOption}
+      name={name}
+      data-h2-position="base(relative)"
+      data-h2-display="base(flex)"
+      data-h2-flex-direction="base(column)"
+      nullable
+    >
+      <div data-h2-position="base(relative)" data-h2-width="base(100%)">
+        <Label hideOptional={hideOptional} required={isRequired}>
+          {label}
+        </Label>
+        <div
+          data-h2-display="base(flex)"
+          data-h2-flex-grow="base(1)"
+          data-h2-width="base(100%)"
+          data-h2-position="base(relative)"
+          data-h2-margin="base(x.125, 0)"
+        >
+          <ComboboxPrimitive.Input
```

Comment:
It might be worth trying suppress the browser's helpers if possible.  I sometimes had my browser take over and capture the up/down key presses.

Note how in this example both are scrolling and then Apache gets selected instead of Adobe.

https://user-images.githubusercontent.com/8978655/207411720-f8470b67-8add-4602-9b11-47b673f5551e.mp4

Reply:
😅 That makes total sense now that you mention it.

Fixed in [disable autocomplete in favour of Combobox.Options](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5064/commits/ee88f49f49329bfc8b43cf2e148dc9fef4cf43be)